src/scripts/combineSplitFilesToMakeALargeDatasetFiles-CSC.ts

```diff
@@ -22,60 +28,119 @@ main.map((item: any, index: number) => {
 	return item;
 });
 
-fs.writeFileSync(path.join(__dirname, '../', 'assets/country.json'), JSON.stringify(allCountries, null, 3));
+fs.writeFileSync(path.join(__dirname, '../', 'assets/allCountryNew.json'), JSON.stringify(allCountries, null, 3));
+
+const countryMetas: { [Property: string]: { countryPath: string } } = allCountries.reduce((accumulator, country) => {
+	accumulator[country.isoCode] = { countryPath: `${country.name.replace(/\W/g, '_')}-${country.isoCode}` };
+	return accumulator;
+}, {} as { [Property: string]: { countryPath: string } });
 
 /**
  * State
  */
-const allStates: any[] = [];
-const alStatesNestedLite = JSON.parse(
-	fs.readFileSync(path.join(__dirname, PATH_TO_DATA_FOLDER, '/allStatesNested.lite.json'), 'utf-8'),
-);
-const alStatesNestedGeo = JSON.parse(
-	fs.readFileSync(path.join(__dirname, PATH_TO_DATA_FOLDER, '/allStatesNested.geo.json'), 'utf-8'),
-);
+const allStates: IState[] = [];
+allCountries.forEach((country) => {
+	process.stdout.write(`Combining ${country.name}-${country.isoCode}...`);
+	const allStatesLite = new Map<string, IState>();
+	const allStatesGeo = new Map<string, IState>();
+	try {
+		const { countryPath } = countryMetas[country.isoCode]; // `${country.name.replace(/\W/g, '_')}-${country.isoCode}`;
+		(JSON.parse(
+			fs.readFileSync(path.join(__dirname, PATH_TO_DATA_FOLDER, countryPath, 'allStates.lite.json'), 'utf-8'),
+		) as IState[]).reduce((accumulator, item) => {
+			accumulator.set(item.isoCode, item);
+			return accumulator;
+		}, allStatesLite);
 
-Object.keys(alStatesNestedLite).map((key) => {
-	alStatesNestedLite[key].map((item: any, index: number) => {
-		const combine = { ...item, ...alStatesNestedGeo[key][index] };
-		allStates.push(combine);
-		return item;
-	});
-	return key;
+		(JSON.parse(
+			fs.readFileSync(path.join(__dirname, PATH_TO_DATA_FOLDER, countryPath, 'allStates.geo.json'), 'utf-8'),
+		) as IState[]).reduce((accumulator, item) => {
+			accumulator.set(item.isoCode, item);
+			return accumulator;
+		}, allStatesGeo);
+
+		[...allStatesLite.keys()].forEach((key) => {
+			assert(allStatesGeo.get(key));
+			const combine: IState = { ...allStatesLite.get(key)!, ...allStatesGeo.get(key)! };
+			allStates.push(combine);
+		});
+	} catch (error) {
+		process.stdout.write(` ${error.message}\n`);
+	} finally {
```

Comment:
getting typescript error - Error is unknown

```js
		if (error instanceof Error) {
			// ✅ TypeScript knows err is Error
			console.log(error.message);
			process.stdout.write(` ${error.message}\n`);
		} else {
			console.log('Unexpected error', error);
		}

```

Reply:
can't reproduce this one... My vscode is using currently typescript 4.9.4, as configured in package.json...
![image](https://user-images.githubusercontent.com/32485627/229627381-11ef8b2f-e2df-4834-bd0f-871efd773911.png)